src/jdk.crypto.mscapi/windows/native/libsunmscapi/security.cpp

```diff
@@ -798,11 +798,14 @@ JNIEXPORT jbyteArray JNICALL Java_sun_security_mscapi_CSignature_signHash
             ::CryptGetProvParam((HCRYPTPROV)hCryptProv, PP_CONTAINER, //deprecated
                 (BYTE *)pbData, &cbData, 0);
 
+            DWORD keysetType;
+            ::CryptGetProvParam((HCRYPTPROV)hCryptProv, PP_KEYSET_TYPE, //deprecated
+                (BYTE*)&keysetType, &cbData, 0);
+
             // Acquire an alternative CSP handle
             if (::CryptAcquireContext(&hCryptProvAlt, LPCSTR(pbData), NULL, //deprecated
-                PROV_RSA_AES, 0) == FALSE)
+                PROV_RSA_AES, 0 | keysetType) == FALSE)
```

Comment:
There is no need for `0 |`.

Reply:
The parameter is a bitmask, the code tries to say `set flags returned by CryptGetProvParam(..., PP_CONTAINER, ...)`.

Reply:
I don't normally add `0 |` since this is not javascript.